lib/puppet/provider/mongodb_user/mongodb.rb

```diff
@@ -103,6 +103,7 @@ def password_hash=(_value)
         pwd: @resource[:password_hash],
         digestPassword: false
       }
+      command[:mechanisms] = @resource[:auth_mechanism] == :scram_sha_1 ? ['SCRAM-SHA-1'] : ['SCRAM-SHA-256']
```

Comment:
'SCRAM-SHA-256' should not be possible.

This mechanism is not compatible with `digestPassword: false`.

@see https://www.mongodb.com/docs/v5.0/reference/command/updateUser/

The update of password_hash is not compatible with 'SCRAM-SHA-256'.

Reply:
Hi @poloz-lab ,
I agree with you, it should not be possible, but the module tries to use 256. I've just tested it with 4.2.0.
1. I've changed password in hiera
2. Run puppet agent and see the nest lines in debug output:

Reply:
Hi @makhovaa ,

Yes you need to change it here.

But I was suggesting to not add the possibility to set 'SCRAM-SHA-256' in your ternary because as you see, we can not change the password when it's SCRAM-SHA-256.

Maybe it would be better to raise an error when we arrive there and we have the SCRAM-SHA-256 mechanism.

What do you think ?

Reply:
I agree, but the problem is I don't try to add SHA-256, The module just try to work with user credentials as if they are SHA-256. After looking through the module code I got the impression SHA-1 is a default mechanism, which must be used if user don't specify the alternative. But on practice, I have different and it tries to apply SHA-256:
`"errmsg" : "Use of SCRAM-SHA-256 requires undigested passwords"`